In [1]:
import kagglehub
import pandas as pd
import os
import joblib
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score

# Download latest version of the dataset
print("Downloading dataset...")
path = kagglehub.dataset_download("johnsmith88/heart-disease-dataset")
print("Path to dataset files:", path)

# Task 1: Data Understanding and Preprocessing
# Load the dataset using Pandas
csv_file_path = os.path.join(path, "heart.csv") # Assuming standard Kaggle filename
df = pd.read_csv(csv_file_path)

# Display the first five records
print("\n--- First 5 Records ---")
print(df.head())

# Identify target variable and numerical features (Assuming 'target' is the label column)
target_col = 'target'
numerical_features = df.select_dtypes(include=['int64', 'float64']).columns.drop(target_col, errors='ignore').tolist()

print(f"\nTarget Variable: {target_col}")
print(f"Numerical Features: {numerical_features}")

# Check for missing values
print("\n--- Missing Values ---")
print(df.isnull().sum())

# Drop rows with missing values (if any exist) or handle them
df = df.dropna()

# Split the dataset into 80% training and 20% testing
X = df.drop(columns=[target_col])
y = df[target_col]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Task 2: Model Development
print("\nTraining the Random Forest model...")
model = RandomForestClassifier(random_state=42, n_estimators=100)
model.fit(X_train, y_train)

# Evaluate using Accuracy Score
y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
print(f"Model Accuracy Score: {accuracy:.4f}")

# Save the trained model using Joblib
joblib.dump(model, 'model.pkl')
print("\nModel saved successfully as 'model.pkl'")

Using Colab cache for faster access to the 'heart-disease-dataset' dataset.
Path to dataset files: /kaggle/input/heart-disease-dataset

--- First 5 Records ---
   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   52    1   0       125   212    0        1      168      0      1.0      2   
1   53    1   0       140   203    1        0      155      1      3.1      0   
2   70    1   0       145   174    0        1      125      1      2.6      0   
3   61    1   0       148   203    0        1      161      0      0.0      2   
4   62    0   0       138   294    1        1      106      0      1.9      1   

   ca  thal  target  
0   2     3       0  
1   0     3       0  
2   0     3       0  
3   1     3       0  
4   3     2       0  

Target Variable: target
Numerical Features: ['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach', 'exang', 'oldpeak', 'slope', 'ca', 'thal']

--- Missing Values ---
age         0
sex         0
cp         

### 1. Setup and Dataset Download

In [2]:
import kagglehub
import pandas as pd
import os
import joblib
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score

# Download latest version of the dataset
print("Downloading dataset...")
path = kagglehub.dataset_download("johnsmith88/heart-disease-dataset")
print("Path to dataset files:", path)

Using Colab cache for faster access to the 'heart-disease-dataset' dataset.
Path to dataset files: /kaggle/input/heart-disease-dataset


### 2. Data Understanding and Preprocessing - Load and Initial View

In [3]:
# Load the dataset using Pandas
csv_file_path = os.path.join(path, "heart.csv") # Assuming standard Kaggle filename
df = pd.read_csv(csv_file_path)

# Display the first five records
print("\n--- First 5 Records ---")
print(df.head())


--- First 5 Records ---
   age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   52    1   0       125   212    0        1      168      0      1.0      2   
1   53    1   0       140   203    1        0      155      1      3.1      0   
2   70    1   0       145   174    0        1      125      1      2.6      0   
3   61    1   0       148   203    0        1      161      0      0.0      2   
4   62    0   0       138   294    1        1      106      0      1.9      1   

   ca  thal  target  
0   2     3       0  
1   0     3       0  
2   0     3       0  
3   1     3       0  
4   3     2       0  


### 3. Identify Target and Features

In [4]:
# Identify target variable and numerical features (Assuming 'target' is the label column)
target_col = 'target'
numerical_features = df.select_dtypes(include=['int64', 'float64']).columns.drop(target_col, errors='ignore').tolist()

print(f"\nTarget Variable: {target_col}")
print(f"Numerical Features: {numerical_features}")


Target Variable: target
Numerical Features: ['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach', 'exang', 'oldpeak', 'slope', 'ca', 'thal']


### 4. Handle Missing Values

In [5]:
# Check for missing values
print("\n--- Missing Values ---")
print(df.isnull().sum())

# Drop rows with missing values (if any exist) or handle them
df = df.dropna()


--- Missing Values ---
age         0
sex         0
cp          0
trestbps    0
chol        0
fbs         0
restecg     0
thalach     0
exang       0
oldpeak     0
slope       0
ca          0
thal        0
target      0
dtype: int64


### 5. Split Data into Training and Testing Sets

In [6]:
# Split the dataset into 80% training and 20% testing
X = df.drop(columns=[target_col])
y = df[target_col]
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

### 6. Model Development and Training (Random Forest)

In [7]:
print("\nTraining the Random Forest model...")
model = RandomForestClassifier(random_state=42, n_estimators=100)
model.fit(X_train, y_train)


Training the Random Forest model...


RandomForestClassifier(random_state=42)

### 7. Model Evaluation

In [8]:
# Evaluate using Accuracy Score
y_pred = model.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
print(f"Model Accuracy Score: {accuracy:.4f}")

Model Accuracy Score: 0.9854


### 8. Save Trained Model

In [9]:
# Save the trained model using Joblib
joblib.dump(model, 'model.pkl')
print("\nModel saved successfully as 'model.pkl'")


Model saved successfully as 'model.pkl'


### Conclusion

This notebook demonstrates a complete workflow for heart disease prediction, from data loading and preprocessing to model training and evaluation. A Random Forest Classifier was trained, achieving a high accuracy score of **0.9854**. The trained model was then saved for future use.